wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php

```diff
@@ -340,52 +340,117 @@ public function getFieldHtml()
+                }
             } else {
-                $value = $data[$this->getObjectProperty()];
-                $method = "variable '" . \get_class($object) . "::$" . $this->getObjectProperty() . "'";
+                $variableName = $this->getObjectProperty();
+                try {
+                    $value = $this->readUploadLocations($data[$variableName] ?? null);
+                } catch (\InvalidArgumentException $e) {
+                    throw new \InvalidArgumentException("The variable {$className}::\${$variableName}UploadFiles() must return an array of strings with the valid file locations for field '{$this->getId()}'.", 0, $e);
+                }
             }
 
-            if (\is_array($value)) {
-                $value = \array_map(function ($v) use ($method) {
-                    if (!\is_string($v) || !\file_exists($v)) {
-                        throw new \InvalidArgumentException(
-                            "The {$method} must return an array of strings with the file locations for field '{$this->getId()}'."
-                        );
-                    }
-
-                    return new UploadFile(
-                        $v,
-                        \basename($v),
-                        ImageUtil::isImage($v, \basename($v), $this->svgImageAllowed()),
-                        true,
-                        $this->svgImageAllowed()
-                    );
-                }, $value);
+            $this->value($value);
+        }
 
-                $this->value($value);
-            } else {
-                throw new \InvalidArgumentException(
-                    "The {$method} must return an array of strings with the file locationsfor field '{$this->getId()}'."
-                );
+        return $this;
+    }
+
+    /**
+     * Reads an array of `UploadFile` objects and validates it.
+     *
+     * @param UploadFile[] $data
+     * @return UploadFile[]
+     * @throws \InvalidArgumentException
+     * @since 5.5
+     */
+    private function readUploadFiles($data): array
+    {
+        if (!\is_array($data)) {
+            throw new \InvalidArgumentException('Given data is not an array.');
+        }
+
+        foreach ($data as $value) {
+            if (!($value instanceof UploadFile)) {
+                throw new \InvalidArgumentException("At least one value is not an instance of '" . UploadFile::class . "'.");
             }
         }
 
-        return $this;
+        return $data;
+    }
+
+    /**
+     * Reads an array of strings containing file locations and validates it.
+     *
+     * @param string[] $data
+     * @return UploadFile[]
+     * @throws \InvalidArgumentException
+     * @since 5.5
+     */
+    private function readUploadLocations($data): array
+    {
+        if (!\is_array($data)) {
+            throw new \InvalidArgumentException('Given data is not an array.');
+        }
```

Comment:
This is better solved with an actual parameter type hint. It might be required to extend the catch-block with `| \TypeError` or `| \ValueError` (or whatever exception PHP throws).